File: query_mongo.js
import { MongoClient } from 'mongodb';
import dotenv from 'dotenv';
dotenv.config();

async function run() {
  const uri = process.env.MONGODB_URI;
  const client = new MongoClient(uri);
  try {
    await client.connect();
    const db = client.db(process.env.MONGODB_DB || "knowledgeVectors");
    const coll = db.collection(process.env.MONGODB_COLLECTION || "knowledge");
    
    const results = await coll.find({ documentId: "BRD Business Reports.docx.pdf" }).sort({ chunkIndex: 1 }).toArray();
    let text = "";
    for (const r of results) {
       if (r.text) text += r.text + "\n\n";
    }
    console.log(text.substring(0, 10000));
  } finally {
    await client.close();
  }
}
run().catch(console.error);

File: query_mongo.py
import os
from pymongo import MongoClient
from dotenv import load_dotenv

load_dotenv()
uri = os.getenv("MONGODB_URI")
db_name = os.getenv("MONGODB_DB", "knowledgeVectors")
coll_name = os.getenv("MONGODB_COLLECTION", "knowledge")

client = MongoClient(uri)
db = client[db_name]
coll = db[coll_name]

# Fetch chunks for the document
chunks = coll.find({"documentId": "BRD Business Reports.docx.pdf"}).sort("chunkIndex", 1)

text_content = ""
for chunk in chunks:
    if "text" in chunk:
        text_content += chunk["text"] + "\n\n"

print(text_content[:4000]) # Print the first 4000 characters to get an idea of the document
print(f"\n--- Total length: {len(text_content)} chars ---")
